pkg/controllers/dynakube/connectioninfo/reconciler.go

```diff
@@ -49,7 +62,23 @@ func (r *Reconciler) Reconcile(ctx context.Context) error {
 		return err
 	}
 
-	return nil
+	needStatusUpdate, err := hasher.IsDifferent(oldStatus.OneAgent.ConnectionInfoStatus, r.dynakube.Status.OneAgent.ConnectionInfoStatus)
+	if err != nil {
+		log.Error(err, "failed to compare OneAgent connection info status hashes")
+		return err
+	} else if needStatusUpdate {
+		err = r.updateDynakubeStatus(ctx)
+		return err
+	}
+
+	needStatusUpdate, err = hasher.IsDifferent(oldStatus.ActiveGate.ConnectionInfoStatus, r.dynakube.Status.ActiveGate.ConnectionInfoStatus)
+	if err != nil {
+		log.Error(err, "failed to compare ActiveGate connection info status hashes")
+		return err
+	} else if needStatusUpdate {
+		err = r.updateDynakubeStatus(ctx)
+	}
+	return err
```

Comment:
No, if the first status is different, it would trigger an update, that implicitly also updates the second one and than does a return, so the second status wouldn't even be checked.
But I agree to your suggestion anyway, it's cleaner.

Reply:
> No, if the first status is different, it would trigger an update, that implicitly also updates the second one and than does a return,

aaaah, true 😵‍💫